opi/__init__.py

```diff
@@ -101,16 +105,25 @@ def add_repo(filename, name, url, enabled=True, gpgcheck=True, gpgkey=None, repo
 	subprocess.call(['sudo', 'cp', tf.name, '/etc/zypp/repos.d/%s.repo' % filename])
 	subprocess.call(['sudo', 'chmod', '644', '/etc/zypp/repos.d/%s.repo' % filename])
 	tf.file.close()
-	refresh_cmd = ['sudo', 'zypper']
-	if auto_import_key:
-		refresh_cmd.append('--gpg-auto-import-keys')
-	refresh_cmd.append('ref')
+	refresh_cmd = []
+	if get_backend() == BackendConstants.zypp:
+		refresh_cmd = ['sudo', 'zypper']
+		if auto_import_key:
+			refresh_cmd.append('--gpg-auto-import-keys')
+		refresh_cmd.append('ref')
+	elif get_backend() == BackendConstants.dnf:
+		refresh_cmd = ['sudo', 'dnf', 'ref']
```

Comment:
Will this automatically import the gpg keys as well?

Reply:
According to https://unix.stackexchange.com/questions/424543/how-to-import-key-that-signs-a-repository-for-dnf-yum it shares the key database with RPM, so `rpm --import` should be enough, and it is called before the refresh happens.